config.go

```diff
@@ -528,6 +532,25 @@ func GetHeadscaleConfig() (*Config, error) {
 			Msgf("'ip_prefixes' not configured, falling back to default: %v", prefixes)
 	}
 
+	oidcClientSecret := viper.GetString("oidc.client_secret")
+	oidcClientSecretPath := viper.GetString("oidc.client_secret_path")
+	if oidcClientSecretPath != "" && oidcClientSecret != "" {
+		return nil, errOidcMutuallyExclusive
+	}
+	if oidcClientSecretPath != "" {
+		secretFile, err := os.Open(os.ExpandEnv(oidcClientSecretPath))
+		if err != nil {
+			return nil, err
+		}
+		defer secretFile.Close()
+
+		secretBytes, err := io.ReadAll(secretFile)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
This can be done in one smooth motion with [`os.ReadFile()`](https://pkg.go.dev/os#ReadFile)

```suggestion
		secretBytes, err := os.ReadFile(os.ExpandEnv(oidcClientSecretPath))
		if err != nil {
			return nil, err
		}
```

I like the use of `os.ExpandEnv`.